onadata/apps/api/management/commands/delete_users.py

```diff
@@ -0,0 +1,116 @@
+"""
+Delete users management command.
+"""
+import sys
+from django.contrib.auth.models import User
+from onadata.apps.logger.models import XForm
+from onadata.apps.logger.models import Instance
+from onadata.apps.logger.models import Project
+from django.core.management.base import BaseCommand
+from django.utils import timezone
+
+
+def get_user_object_stats(username):  # pylint: disable=R0201
+    """
+    Get User information.
+    """
+    user = User.objects.get(username=username)
+    # Get the number of projects for this user
+    user_projects = Project.objects.filter(created_by=user).count()
+    # Get the number of forms
+    user_forms = XForm.objects.filter(user=user).count()
+    # Get the number of submissions
+    user_sumbissions = Instance.objects.filter(user=user).count()
+
+    result = {
+        'projects': user_projects,
+        'forms': user_forms,
+        'submissions': user_sumbissions
+    }
+
+    return result
+
+
+def inactivate_user(username):
+    """
+    Soft deletes the user termporarily.
+    """
+    try:
+        user = User.objects.get(username=username)
+        # set inactive status on user account
+        user.is_active = False
+        # append a timestamped suffix to the username
+        # to make the initial username available
+        soft_deletion_time = timezone.now()
+        deletion_suffix = soft_deletion_time.strftime('-deleted-at-%s')
+        user.username += deletion_suffix
+
+        user.save()
+        sys.stdout.write('User {} deleted successfully.'.format(username))
+        # confirm too that no user exists with provided email
+        try:
+            User.objects.get(email=user.email).delete()
+        except User.DoesNotExist:
+            pass
+    except User.DoesNotExist:
+        sys.stdout.write('User {} does not exist.'.format(username))
```

Comment:
Any scenario where we would need to use this somewhere else in onadata ? I think it would be nice to move these functions into a common area...

Reply:
Well, I got the concept of soft deleting from the Instance model.
Seems soft deletion occurs in the respective models i.e Form, Project, Submission.
Would we really want to streamline all those functions to use a collective function?

Reply:
Mhh no... we probably don't need to streamline them all.... I think they each do something different ? Might be worth checking out in the future.

Reply:
Mhh I guess it's fine to leave these functions here... if we won't need to run these anywhere else....

_I'd still suggest to move them to a `lib` folder but it's not really necessary_